addons/hr_expense/models/hr_expense.py

```diff
@@ -868,14 +868,14 @@ def action_sheet_move_create(self):
         expense_line_ids = self.mapped('expense_line_ids')\
             .filtered(lambda r: not float_is_zero(r.total_amount, precision_rounding=(r.currency_id or self.env.company.currency_id).rounding))
         res = expense_line_ids.action_move_create()
+        for sheet in self:
+            if sheet.accounting_date:
+                sheet.accounting_date = sheet.account_move_id.date
 
-        if not self.accounting_date:
-            self.accounting_date = self.account_move_id.date
-
-        if self.payment_mode == 'own_account' and expense_line_ids:
-            self.write({'state': 'post'})
-        else:
-            self.write({'state': 'done'})
+            if sheet.payment_mode == 'own_account' and sheet.expense_line_ids:
+                sheet.write({'state': 'post'})
+            else:
+                sheet.write({'state': 'done'})
```

Comment:
You can batch the writes like below.
As the action might be done on a lot of records at once, the ORM will not have to write on each record individually.

```suggestion
        to_post = self.filtered(lambda sheet: sheet.payment_mode == 'own_account' and sheet.expense_line_ids)
        to_post.write({'state': 'post'})
        (self - to_post).write({'state': 'done'})
```